go.mod

```diff
@@ -1,6 +1,8 @@
 module github.com/launchdarkly/terraform-provider-launchdarkly
 
-go 1.21
+go 1.22.9
```

Comment:
If you're going to update go here then I think we should also update it everywhere else (e.g. in the GHA and in the `.go-version` file. However, maybe this should be a separate PR.

Reply:
@ldhenry https://github.com/launchdarkly/terraform-provider-launchdarkly/pull/284